flink-connector-mongodb/src/main/java/org/apache/flink/connector/mongodb/source/MongoSourceBuilder.java

```diff
@@ -0,0 +1,219 @@
+        connectionOptionsBuilder.setUri(uri);
+        return this;
+    }
+
+    /**
+     * Sets the database to sink of MongoDB.
+     *
+     * @param database the database to sink of MongoDB.
+     * @return this builder
+     */
+    public MongoSourceBuilder<OUT> setDatabase(String database) {
+        connectionOptionsBuilder.setDatabase(database);
+        return this;
+    }
+
+    /**
+     * Sets the collection to sink of MongoDB.
+     *
+     * @param collection the collection to sink of MongoDB.
+     * @return this builder
+     */
+    public MongoSourceBuilder<OUT> setCollection(String collection) {
+        connectionOptionsBuilder.setCollection(collection);
+        return this;
+    }
+
+    /**
+     * Sets the number of documents should be fetched per round-trip when reading.
+     *
+     * @param fetchSize the number of documents should be fetched per round-trip when reading.
+     * @return this builder
+     */
+    public MongoSourceBuilder<OUT> setFetchSize(int fetchSize) {
+        readOptionsBuilder.setFetchSize(fetchSize);
+        return this;
+    }
+
+    /**
+     * Sets the batch size of MongoDB find cursor.
+     *
+     * @param cursorBatchSize the max batch size of find cursor.
+     * @return this builder
+     */
+    public MongoSourceBuilder<OUT> setCursorBatchSize(int cursorBatchSize) {
+        readOptionsBuilder.setCursorBatchSize(cursorBatchSize);
+        return this;
+    }
+
+    /**
+     * Set this option to true to prevent cursor timeout (defaults to 10 minutes).
+     *
+     * @param noCursorTimeout Set this option to true to prevent cursor timeout.
+     * @return this builder
+     */
+    public MongoSourceBuilder<OUT> setNoCursorTimeout(boolean noCursorTimeout) {
+        readOptionsBuilder.setNoCursorTimeout(noCursorTimeout);
+        return this;
+    }
+
+    /**
+     * Sets the partition strategy.
+     *
+     * @param partitionStrategy the strategy of a partition.
+     * @return this builder
+     */
+    public MongoSourceBuilder<OUT> setPartitionStrategy(PartitionStrategy partitionStrategy) {
+        readOptionsBuilder.setPartitionStrategy(partitionStrategy);
+        return this;
+    }
+
+    /**
+     * Sets the partition size of MongoDB split.
+     *
+     * @param partitionSize the memory size of a partition.
+     * @return this builder
+     */
+    public MongoSourceBuilder<OUT> setPartitionSize(MemorySize partitionSize) {
+        readOptionsBuilder.setPartitionSize(partitionSize);
+        return this;
+    }
+
+    /**
+     * Sets the samples size per partition only effective for sample partition strategy.
+     *
+     * @param samplesPerPartition the samples size per partition
+     * @return this builder
+     */
+    public MongoSourceBuilder<OUT> setSamplesPerPartition(int samplesPerPartition) {
+        readOptionsBuilder.setSamplesPerPartition(samplesPerPartition);
+        return this;
+    }
+
+    /**
+     * Sets the limit of documents to read.
```

Comment:
What happens if limit is not set?

Reply:
How about change it to the following?
```
Sets the limit of documents to read. If limit is not set or set to -1, we will not push the limit down to MongoDB.
```

Reply:
What does it mean to not push the limit to MongoDB?